larq_compute_engine/core/bgemm_kernels_arm32.h

```diff
@@ -0,0 +1,527 @@
+#ifndef COMPUTE_EGNINE_TFLITE_KERNELS_BGEMM_KERNELS_ARM_H_
+#define COMPUTE_EGNINE_TFLITE_KERNELS_BGEMM_KERNELS_ARM_H_
+
+#include <cstdint>
+
+#include "bgemm_kernels_common.h"
+#include "ruy/profiler/instrumentation.h"
+
+using namespace ruy;
+
+#if RUY_PLATFORM_NEON && RUY_OPT(ASM) && RUY_PLATFORM_NEON_32
+
+// clang-format off
+
+#define LCE_BMLA(Vd, Vr, Vl0, Vl1, Vl2, Vl3) \
+  "veor.s8 q12, " #Vr", " #Vl0 "\n"    \
+  "veor.s8 q13, " #Vr", " #Vl1 "\n"    \
+  "veor.s8 q14, " #Vr", " #Vl2 "\n"    \
+  "veor.s8 q15, " #Vr", " #Vl3 "\n"    \
+  "vcnt.s8 q12, q12\n"                 \
+  "vcnt.s8 q13, q13\n"                 \
+  "vcnt.s8 q14, q14\n"                 \
+  "vcnt.s8 q15, q15\n"                 \
+  "vpadd.i8 d24, d24, d25\n"           \
+  "vpadd.i8 d25, d26, d27\n"           \
+  "vpadd.i8 d28, d28, d29\n"           \
+  "vpadd.i8 d29, d30, d31\n"           \
+  "vpadd.i8 d24, d24, d25\n"           \
+  "vpadd.i8 d25, d28, d29\n"           \
+  "vpaddl.s8 q12, q12\n"               \
+  "vpadal.s16 " #Vd" , q12\n"
```

Comment:
Perhaps these should use unsigned datatypes?

```suggestion
  "vpaddl.u8 q12, q12\n"               \
  "vpadal.u16 " #Vd" , q12\n"
```

(I appreciate that it doesn't make a difference to the computation in this case because the maximal value in each byte even after the `vpadd` instructions is < 2^7, but I think it's slightly nicer.)